contributor/04-mechanics-of-contributing.md

```diff
@@ -0,0 +1,125 @@
+(This might actually be pretty similar to what you might already be doing on your own project to get your changes accepted - if that's the case: great, this is going to come natural to you!)
+
+### Testing and automation
+
+The basic point here is to enable the trusted committer to validate the contribution without your presence and to ensure easy maintainability.
+To achieve this do the following: 
+
+- Add tests for your code contribution so that validating the function of your contribution by others works well, even after some time when you work in other projects or might have stopped to contribute to this project.
+  - Often projects will use automated checks against pull request using those tests and the level of code coverage. Try to meet the criteria these test enforce.
+- Many projects will provide project build and validations scripts to enable you to locally test your changes. 
+  - Use those to ensure that your contribution works as well as possible before opening a pull request. 
+  - Having to review defect pull requests with easy-to-fix errors often bugs trusted committers. They will not fix your code but ask you to do so. This might create more round-trips and slow the merge if there's unnecessary errors.
+  - No one's perfect though. Try your best, use the prepared validation scripts if there are any and give it your best shot with a pull request!
+- Don't forget your own project that triggered your contribution in the first place. Build a modified build of the shared project with your changes and try it out in your own project that consumes it.
+
+### Documentation and reviewability
+
+You'll want to ensure that your pull request includes any documentation updates are relevant to your changes. 
+Should the documentation live in a different place, make sure you add them there and link to them in the pull request.
+
+To make the actual code review as easy as possible for the trusted committer or other persons reviewing it, try to follow these hints:
+
+- Be sure that your pull request includes just the relevant changes for the issue you're completing.
+- Try to avoid super-large commits, commits with unclear commit messages, gazillions of files, incoherent changes (e.g. touching multiple topics).
+- Provide a clear description what this pull request changes, why it does so and which issue and possible design documents (if there were any) it refers to.
+- If there is anything uncommon or unexpected in the pull request highlight it and provide an explanation. This will make it easier to reason about and solve potential blocking questions the reviewer might have during the review. 
+  - The same goes for place where you were unsure in the implementation or your approach - highlight it and ask for insight.
+  - Be civil and expect civility from the trusted committers' review.   
+- Making pull requests too broad and large makes them more difficult to review and longer before they're accepted.
```

Comment:
```suggestion
- Making pull requests too broad and large makes them more difficult to review so it will take much longer before they're accepted.
```